CONTRIBUTING.rst

```diff
@@ -255,15 +255,15 @@ Guidelines
           # function not for directed graphs *or* for multigraphs
           pass
 
```

Comment:
@MridulS , @rossbar I rearranged /edited this section.  Take another look to see if you are still OK w/ it.

Reply:
Here it is rendered:
https://305-890377-gh.circle-artifacts.com/0/doc/build/html/developer/contribute.html#guidelines

I wonder if we should rearrange the guidelines to have the dependency thing earlier.

Like this
---

* All code should have tests.
* All code should be documented, to the same
  `standard <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#docstring-standard>`_
  as NumPy and SciPy.
* All changes are reviewed.  Ask on the
  `mailing list <http://groups.google.com/group/networkx-discuss>`_ if
  you get no response to your pull request.
* Default dependencies are listed in ``requirements/default.txt`` and extra
  (i.e., optional) dependencies are listed in ``requirements/extra.txt``.
  We don't often add new default and extra dependencies.  Typically, new
  proposed dependencies would first be added as extra dependencies.
  Extra dependencies should be easy to install on all platforms and widely-used.
  New default dependencies should be easy to install on all platforms,
  widely-used in the community, and have demonstrated potential for wide-spread
  use in NetworkX.
* Use the following import conventions::

   import numpy as np
   import scipy as sp
   import matplotlib as mpl
   import matplotlib.pyplot as plt
   import pandas as pd
   import networkx as nx

* Use the decorator ``not_implemented_for`` in ``networkx/utils/decorators.py``
  to designate that a function doesn't accept 'directed', 'undirected',
  'multigraph' or 'graph'.  The first argument of the decorated function should
  be the graph object to be checked.

  .. code-block:: python
  
      @nx.not_implemented_for('directed', 'multigraph')
      def function_not_for_MultiDiGraph(G, others):
          # function not for graphs that are directed *and* multigraph
          pass

      @nx.not_implemented_for('directed')
      @nx.not_implemented_for('multigraph')
      def function_only_for_Graph(G, others):
          # function not for directed graphs *or* for multigraphs
          pass

Reply:
LGTM.